### PR TITLE
Update callers of FindGridSpots

### PR DESCRIPTION
**Pull request by effting**
Initial commit - to be tested and discussed

### CODE REVIEW

File: scripts/spot-grid.py [hunk 1]
```diff
@@ -53,12 +58,22 @@ def new_image(self, data):
...
+            data,
+            (8, 8),
+            AffineTransform,
+            sigma=1.45,
+            threshold_rel=0.5,
+        )
+        grid = unit_gridpoints((8, 8), mode="ji")
+        self.app.spots = tform_ji.apply(grid)
+
+        tform_xy = to_physical_space(tform_ji, data.shape, pxsize)
```
Comment (tepals) [inline scripts/spot-grid.py:71]: `to_physical_space` needs to be imported and pxsize is not defined here, can be solved by setting self.pxsize in the init

Comment (tepals) [reply scripts/spot-grid.py:71]: I think there is also still a bug in the `to_physical_space` for a transform. I checked this code and `tform_xy.scale` is exactly equal to `tform_ji.scale`

**Comment by effting** [comment]
I've updated the PR to something that works. Please give it a review.
I suggest we also check this on EA2 before merging

File: scripts/spot-grid.py [hunk 2]
```diff
@@ -53,12 +61,30 @@ def new_image(self, data):
...
         """
-        self.app.spots, trans, scale, rot, shear = FindGridSpots(data, self.gridsize)
-        self.app.translation = trans[0], data.shape[0] - trans[1]
-        self.app.scale = scale
-        self.app.rotation = -rot
-        self.app.shear = shear
-
+        tform_ji, _ = estimate_grid_orientation_from_img(
+            data,
+            (8, 8),
```
Comment (tepals) [inline scripts/spot-grid.py:66]: you can use `self.gridsize` instead of (8, 8), to allow compatibility with images that have grids with different sizes.

**Review by tepals — CHANGES_REQUESTED** [review]

File: scripts/spot-grid.py [hunk 3]
```diff
@@ -53,12 +61,30 @@ def new_image(self, data):
...
+        self.app.spots = tform_ji.apply(grid)
+
+        tform_xy = functools.reduce(
+            operator.matmul,
+            (
+                to_physical_space_transform(data.shape, self.app.pixel_size),
+                tform_ji,
+                to_pixel_index_transform(),
+            ),
+        )
```
[the remark this answers is not in the document]
Comment (pieleric) [reply scripts/spot-grid.py:81]: The whole point of operators is to write them, like `@`. Now, for this current code, you have to check how `reduce` work, and read the whole `matmul` word, and realise that the next 3 arguments are actually 3 transforms. The explicit version is much easier to read.

[the remark this answers is not in the document]
Comment (effting) [reply scripts/spot-grid.py]: OK - changed it

File: scripts/spot-grid.py (hunk 2 above)
Comment (effting) [reply scripts/spot-grid.py]: thanks

**Review by pieleric — APPROVED** [review]

**Review by tepals — APPROVED** [review]

**Pull request merged**